Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs

```diff
@@ -48,6 +49,14 @@ void Move(NotifyCollectionChangedEventArgs args)
 
 			_collectionView.MoveItem(oldPath, newPath);
 		}
+		
+		private void Replace(NotifyCollectionChangedEventArgs args)
+		{
+			var startIndex = args.NewStartingIndex > -1 ? args.NewStartingIndex : _itemsSource.IndexOf(args.NewItems[0]);
```

Comment:
Huh. Why don't we throw here? What does a negative index mean? What does it mean if the `NewStartingIndex` is not the index of the first element of `NewItems`? Isn't that what it means to replace? That the index is the same for the old and new item? Anyway, I'm sure there's a good reason but for future spelunkers a note would be nice.